### ISSUE TITLE
Commercial update to typo3 v8?

### ISSUE THREAD

**Issue by kftw** [YAG-Gallery/yag#121]
Hi! Is there a chance you could update YAG to TYPO3 v8 compatibillity if payed? We have some scenarios where a migration to other plugins isn't suitable. 

Cheers!

**Comment by daniellienert** [comment]
Hey @kftw, 
as said, we would do the update, as soon as it is needed in one of our 8.x projects. That didn't happen so far. If you want to support such an upgrade please contact me directly via mail or slack to discuss the details.

Cheers,
Daniel

**Comment by kftw** [comment]
Yeah, we would really be interessted in that. Did you receive my email? I could find one here so i used your @lienert.cc address from forge.typo3.org. Or whats your slack channel?

**Comment by nvindice** [comment]
@kftw, @daniellienert Just curious... Is there any progress to expect? I'm operating a small non-profit page and would like to know, if and when I can upgrade to T3 v8 without having to find a new gallery (...and customizing it etc pp).

Thanks!

**Comment by kftw** [comment]
@nvindice Unfortunately it looks like it won't happen. For our project we'll write some migration scripts. 

The first step will be to implement one or two scheduler tasks to migrate all relevant data from tx_yag_domain_model_item and tx_yag_domain_model_itemmeta to sys_file and an extended sys_file_metadata. 

From there one can, for example, create new  image gallery content elements and feed them with the migrated data.

If you're interested and our effort leads to something usable, I can upload it here.

**Comment by nvindice** [comment]
@kftw Thanks for the lightning fast response, and I'm definitely interested in your progress. Sad to hear the plugin wont be migrated to v8 anytime soon... Did you choose a v8-compatible gallery extension yet?

**Comment by kftw** [comment]
@nvindice Yeah, it's sad for this great extension. On the other hand - many things on the image handling side now are core features of TYPO3. 

We haven't decided yet, but it's probably going to be something simple.

**Comment by wrock** [comment]
@kftw Did you have success with your migrations scripts? Can you maybe share them?

**Comment by instruccionesaldorso** [comment]
@kftw I would be interested regarding that migration too.
We are not able to make the extension work in TYPO3 7.6 (dependencies with EXT:pt_extbase and EXT:pt_extlist don't allow to display the gallery on the frontend).

**Comment by wrock** [comment]
@instruccionesaldorso in 7.6 it should work. Maybe you took the latest version of pt_extbase and pt_extlist. You have to take the two dependencies in the right versions which are referred to in yag.

**Comment by instruccionesaldorso** [comment]
@wrock This is what we install:
- Installing typo3-ter/pt-extbase (2.3.0): Loading from cache
- Installing typo3-ter/pt-extlist (2.4.0): Loading from cache
- Installing typo3-ter/yag (4.2.1): Loading from cache

Which versions do you have exactly?
Are you installing EXT:yag through composer?
In that case, what do you use: "typo3-ter/yag" or "dl/yag"?

**Comment by wrock** [comment]
Dependencies for YAG 4.2.1 (https://extensions.typo3.org/extension/yag/)
pt_extlist (2.0.4)
pt_extbase (2.0.0)
TYPO3 (7.6.0 - 7.6.99)

We used the extension-manager.

**Comment by instruccionesaldorso** [comment]
I guess you jumped the dependencies when installing EXT:pt_extbase and EXT:pt_extlist, because both  versions you mentioned depend on TYPO3 <= 7.0.99.
I have an up-to-date 7.6 instance.
I tried a lot of version combinations (and other forks), with no success.

**Comment by kftw** [comment]
@wrock @instruccionesaldorso Sorry, we didn't have much success with automatic migration because each of the existing galleries had differenct configurations. So we settled with manually creating new galleries :-(

**Comment by instruccionesaldorso** [comment]
@kftw Just for curiosity, you've just used sys_file records and text+media elements, overwritting EXT:fluid_styled_content templates, or you just used another third-party extension from TER?

**Comment by Bunnyfield** [comment]
We are currently working on a fully namespaced version of pt_extbase, pt_extlist and yag for CMS 8, so if some of you people are still interested in a commercial upgrade, feel free to contact me for some sponsoring.
